v2/lockrenewer.go

```diff
@@ -170,3 +179,35 @@ func (plr *peekLockRenewer) startPeriodicRenewal(ctx context.Context, message *a
 		}
 	}
 }
+
+func (plr *peekLockRenewer) renewMessageLock(ctx context.Context, message *azservicebus.ReceivedMessage, options *azservicebus.RenewMessageLockOptions) error {
+	lockLostErr := &azservicebus.Error{Code: azservicebus.CodeLockLost}
+	renewalTimeout := time.Until(*message.LockedUntil) // assumes messages.LockedUntil is set
```

Comment:
I think that's fine, but if the middleware is used with Peek, that might crash.
it wouldn't make sense anyways to have this renewer for a message that isn't locked. so I'm fine with crashing here

Reply:
added a check here to return lock lost if there is no LockUntil